pkg/event/event.go

```diff
@@ -452,9 +512,13 @@ func (e *EventHandler) updateCLockClass(cfgName string, ptpState PTPState, clock
 	case PTP_HOLDOVER:
 		switch clockType {
 		case GM:
-			// Need to know inSpec or Outof Spec
-			// update only when ClockClass is changed
-			if g.ClockQuality.ClockClass != fbprotocol.ClockClass7 {
+			if e.isSourceLost(cfgName) && !e.outOfSpec && g.ClockQuality.ClockClass != fbprotocol.ClockClass7 {
+				g.ClockQuality.ClockClass = fbprotocol.ClockClass7
+				g.ClockQuality.ClockAccuracy = fbprotocol.ClockAccuracyUnknown
+				// T-REC-G.8275.1-202211-I section 6.3.5
+				g.ClockQuality.OffsetScaledLogVariance = 0xffff
+				err = gmSetterFn(cfgName, g)
+			} else if e.isSourceLost(cfgName) && e.outOfSpec && g.ClockQuality.ClockClass != protocol.ClockClassOutOfSpec {
 				g.ClockQuality.ClockClass = fbprotocol.ClockClass7
```

Comment:
should be 
`g.ClockQuality.ClockClass = protocol.ClockClassOutOfSpec`

Reply:
review again pls